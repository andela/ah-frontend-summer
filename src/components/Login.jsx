import React from "react";
import PropTypes from 'prop-types';
import { Form, Grid, Segment, Message, Divider, Header } from "semantic-ui-react";
import GoogleLogin from "react-google-login";
import FacebookLogin from "react-facebook-login";
import { NavLink } from 'react-router-dom';
import config from "../config";
import classes from "../styles/login.module.css";
import '../styles/login.css';


const Login = props => {
    const {
        email,
        password,
        onDismissHandler,
        loginError,
        onSubmitHandler,
        onChangeHandler,
        loading,
        onBlurHandler,
        isEmailValid,
        responseFacebook,
        responseGoogle,
        onFailure
    } = props;

    const loginErorMsg = (
        <Message
            negative
            content={loginError}
            onDismiss={onDismissHandler}
            style={{ width: "50%", margin: "auto" }}
        />
    );
    const emailInvalidError = <div style={{ color: "red" }}>Please enter a valid email</div>;
    return (
        <div className={classes.LoginPage}>
            <Grid columns={2} stackable>
                <Grid.Column width={7}></Grid.Column>
                <Grid.Column
                    width={8}
                    padded="very"
                >
                    <Segment padded="very" color="green">

                        <Header as="h1" textAlign="center" padded="very">Author&apos;s Haven</Header>
                        <Header as="h6" style={{ color: "#008080", marginTop: "5px", textAlign: "center", fontSize: "1em" }}>A home for the creative at heart</Header>

                        <br />
                        {loginError ? loginErorMsg : null}
                        <Form onSubmit={onSubmitHandler} className={classes.cardForm}>
                            <center>
                                <Form.Field>
                                    <input
                                        style={!isEmailValid ? { borderColor: "red", width: "50%" } : { width: "50%" }}
                                        type="text"
                                        placeholder="Email"
                                        name="email"
                                        value={email}
                                        onChange={onChangeHandler}
                                        onBlur={onBlurHandler}
                                    />
                                    {!isEmailValid ? emailInvalidError : null}
                                </Form.Field>
                                <Form.Field>
                                    <input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={password}
                                        onChange={onChangeHandler}
                                        style={{ width: "50%" }}
                                    />
                                </Form.Field>
                                <Form.Button type="submit" className="login" loading={loading}>Sign in</Form.Button>
                            </center>
                        </Form>
                        <Divider horizontal>Or</Divider>
                        <br />
                        <div className={classes.divide}>
                            <center>
                                <GoogleLogin
                                    className={classes.googleButton}
                                    clientId={config.GOOGLE_CLIENT_ID}
                                    onSuccess={responseGoogle}
                                    onFailure={onFailure}
                                />
                                <br />
                                <br />

                                <FacebookLogin
                                    appId={config.FACEBOOK_CLIENT_ID}
                                    autoLoad={false}
                                    fields="name,email,picture"
                                    cssClass={classes.faceBookButton}
                                    callback={responseFacebook}
                                />
                            </center>
                        </div>
                        <br />
                        <Divider horizontal>Or</Divider>

                        <div className="spaceComponents">
                            <NavLink to="/signup">
                                <Header style={{ color: "#008080" }} textAlign="center" padded="very">Sign up</Header>
                            </NavLink>
                        </div>

                    </Segment>
                </Grid.Column>
            </Grid>
        </div>
    );
};

Login.propTypes = {
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    onDismissHandler: PropTypes.func.isRequired,
    loginError: PropTypes.string.isRequired,
    onSubmitHandler: PropTypes.func.isRequired,
    onChangeHandler: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    onBlurHandler: PropTypes.func.isRequired,
    responseFacebook: PropTypes.func.isRequired,
    responseGoogle: PropTypes.func.isRequired,
    isEmailValid: PropTypes.bool.isRequired
};

export default Login;
