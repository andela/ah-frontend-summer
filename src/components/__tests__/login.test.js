import React from "react";
import { mount } from "enzyme";
import { BrowserRouter } from 'react-router-dom';
import Login from "../Login";

describe("Login", () => {
    const onChangeHandlerMock = jest.fn();
    const onSubmitHandlerMock = jest.fn();
    const responseFacebookMock = jest.fn();
    const responseGoogleMock = jest.fn();
    const onBlurHandlerMock = jest.fn();
    
    const loginProps = {
        onChangeHandler: onChangeHandlerMock,
        loginError: "whsdhfish",
        onSubmitHandler: onSubmitHandlerMock,
        email: "",
        password: "",
        onDismissHandler: jest.fn(),
        loading: false,
        responseFacebook: responseFacebookMock,
        responseGoogle: responseGoogleMock,
        onBlurHandler: onBlurHandlerMock,
        isEmailValid: true
    };
    const login = mount(<BrowserRouter><Login {...loginProps} /></BrowserRouter>);

    it("renders login component correctly", () => {
        expect(login).toMatchSnapshot();
    });

    it("calls onChandeHandler when the email or password input is changed", () => {
        login.find('input[name="email"]').simulate("change");
        expect(onChangeHandlerMock).toHaveBeenCalled();
        login.find('input[name="password"]').simulate("change");
        expect(onChangeHandlerMock).toHaveBeenCalled();
    });

    it("renders Message component when loginError is true", () => {
        expect(login.find("Message").exists()).toBe(true);
    });

    it("calls onSubmitHandler when form is submited", () => {
        login.find("form").simulate("submit");
        expect(onSubmitHandlerMock).toHaveBeenCalled();
    });

});
