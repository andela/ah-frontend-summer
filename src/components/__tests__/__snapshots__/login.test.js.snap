// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders login component correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login
      email=""
      isEmailValid={true}
      loading={false}
      loginError="whsdhfish"
      onBlurHandler={[MockFunction]}
      onChangeHandler={[MockFunction]}
      onDismissHandler={[MockFunction]}
      onSubmitHandler={[MockFunction]}
      password=""
      responseFacebook={[MockFunction]}
      responseGoogle={[MockFunction]}
    >
      <div
        className="LoginPage"
      >
        <Grid
          columns={2}
          stackable={true}
        >
          <div
            className="ui stackable two column grid"
          >
            <GridColumn
              width={7}
            >
              <div
                className="seven wide column"
              />
            </GridColumn>
            <GridColumn
              padded="very"
              width={8}
            >
              <div
                className="eight wide column"
                padded="very"
              >
                <Segment
                  color="green"
                  padded="very"
                >
                  <div
                    className="ui green very padded segment"
                  >
                    <Header
                      as="h1"
                      padded="very"
                      textAlign="center"
                    >
                      <h1
                        className="ui center aligned header"
                        padded="very"
                      >
                        Author's Haven
                      </h1>
                    </Header>
                    <Header
                      as="h6"
                      style={
                        Object {
                          "color": "#008080",
                          "fontSize": "1em",
                          "marginTop": "5px",
                          "textAlign": "center",
                        }
                      }
                    >
                      <h6
                        className="ui header"
                        style={
                          Object {
                            "color": "#008080",
                            "fontSize": "1em",
                            "marginTop": "5px",
                            "textAlign": "center",
                          }
                        }
                      >
                        A home for the creative at heart
                      </h6>
                    </Header>
                    <br />
                    <Message
                      content="whsdhfish"
                      negative={true}
                      onDismiss={[MockFunction]}
                      style={
                        Object {
                          "margin": "auto",
                          "width": "50%",
                        }
                      }
                    >
                      <div
                        className="ui negative message"
                        style={
                          Object {
                            "margin": "auto",
                            "width": "50%",
                          }
                        }
                      >
                        <Icon
                          as="i"
                          name="close"
                          onClick={[Function]}
                        >
                          <i
                            aria-hidden="true"
                            className="close icon"
                            onClick={[Function]}
                          />
                        </Icon>
                        <MessageContent>
                          <div
                            className="content"
                          >
                            <p>
                              whsdhfish
                            </p>
                          </div>
                        </MessageContent>
                      </div>
                    </Message>
                    <Form
                      as="form"
                      className="cardForm"
                      onSubmit={[MockFunction]}
                    >
                      <form
                        className="ui form cardForm"
                        onSubmit={[Function]}
                      >
                        <center>
                          <FormField>
                            <div
                              className="field"
                            >
                              <input
                                name="email"
                                onBlur={[MockFunction]}
                                onChange={[MockFunction]}
                                placeholder="Email"
                                style={
                                  Object {
                                    "width": "50%",
                                  }
                                }
                                type="text"
                                value=""
                              />
                            </div>
                          </FormField>
                          <FormField>
                            <div
                              className="field"
                            >
                              <input
                                name="password"
                                onChange={[MockFunction]}
                                placeholder="Password"
                                style={
                                  Object {
                                    "width": "50%",
                                  }
                                }
                                type="password"
                                value=""
                              />
                            </div>
                          </FormField>
                          <FormButton
                            as={[Function]}
                            className="login"
                            control={[Function]}
                            loading={false}
                            type="submit"
                          >
                            <FormField
                              className="login"
                              control={[Function]}
                              loading={false}
                              type="submit"
                            >
                              <div
                                className="field login"
                              >
                                <Button
                                  as="button"
                                  loading={false}
                                  type="submit"
                                >
                                  <button
                                    className="ui button"
                                    onClick={[Function]}
                                    type="submit"
                                  >
                                    Sign in
                                  </button>
                                </Button>
                              </div>
                            </FormField>
                          </FormButton>
                          <Link
                            replace={false}
                            to="/forgot-password"
                          >
                            <a
                              href="/forgot-password"
                              onClick={[Function]}
                            >
                              Forgot Password?
                            </a>
                          </Link>
                        </center>
                      </form>
                    </Form>
                    <Divider
                      horizontal={true}
                    >
                      <div
                        className="ui horizontal divider"
                      >
                        Or
                      </div>
                    </Divider>
                    <br />
                    <div
                      className="divide"
                    >
                      <center>
                        <e
                          accessType="online"
                          buttonText="Sign in with Google"
                          className="googleButton"
                          clientId="134825392968-d9p3skmcrstu1q0t8ll6nhikphc8ovjg.apps.googleusercontent.com"
                          cookiePolicy="single_host_origin"
                          disabledStyle={
                            Object {
                              "opacity": 0.6,
                            }
                          }
                          fetchBasicProfile={true}
                          icon={true}
                          isSignedIn={false}
                          jsSrc="https://apis.google.com/js/api.js"
                          onRequest={[Function]}
                          onSuccess={[MockFunction]}
                          prompt=""
                          scope="profile email"
                          tag="button"
                          theme="light"
                          type="button"
                          uxMode="popup"
                        >
                          <button
                            className="googleButton"
                            disabled={true}
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#fff",
                                "border": "1px solid transparent",
                                "borderRadius": 2,
                                "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                                "color": "rgba(0, 0, 0, .54)",
                                "display": "inline-flex",
                                "fontFamily": "Roboto, sans-serif",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "opacity": 0.6,
                                "padding": 0,
                              }
                            }
                            type="button"
                          >
                            <Component
                              active={false}
                              key="1"
                            >
                              <div
                                style={
                                  Object {
                                    "background": "#fff",
                                    "borderRadius": 2,
                                    "marginRight": 10,
                                    "padding": 10,
                                  }
                                }
                              >
                                <svg
                                  height="18"
                                  width="18"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <g
                                    fill="#000"
                                    fillRule="evenodd"
                                  >
                                    <path
                                      d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                                      fill="#EA4335"
                                    />
                                    <path
                                      d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                                      fill="#4285F4"
                                    />
                                    <path
                                      d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                                      fill="#FBBC05"
                                    />
                                    <path
                                      d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                                      fill="#34A853"
                                    />
                                    <path
                                      d="M0 0h18v18H0z"
                                      fill="none"
                                    />
                                  </g>
                                </svg>
                              </div>
                            </Component>
                            <b
                              icon={true}
                              key="2"
                            >
                              <span
                                style={
                                  Object {
                                    "fontWeight": 500,
                                    "paddingBottom": 10,
                                    "paddingLeft": 0,
                                    "paddingRight": 10,
                                    "paddingTop": 10,
                                  }
                                }
                              >
                                Sign in with Google
                              </span>
                            </b>
                          </button>
                        </e>
                        <br />
                        <br />
                        <t
                          appId={229135374703025}
                          autoLoad={false}
                          callback={[MockFunction]}
                          cssClass="faceBookButton"
                          fields="name,email,picture"
                          size="metro"
                          tag="button"
                          textButton="Login with Facebook"
                          typeButton="button"
                        >
                          <t
                            appId={229135374703025}
                            authType=""
                            autoLoad={false}
                            callback={[MockFunction]}
                            cookie={false}
                            cssClass="faceBookButton"
                            disableMobileRedirect={false}
                            fields="name,email,picture"
                            isMobile={false}
                            language="en_US"
                            onFailure={null}
                            redirectUri="http://localhost/"
                            render={[Function]}
                            responseType="code"
                            returnScopes={false}
                            scope="public_profile,email"
                            size="metro"
                            state="facebookdirect"
                            tag="button"
                            textButton="Login with Facebook"
                            typeButton="button"
                            version="2.3"
                            xfbml={false}
                          >
                            <span
                              style={
                                Object {
                                  "opacity": 0.6,
                                  "transition": "opacity 0.5s",
                                }
                              }
                            >
                              <button
                                className="faceBookButton metro"
                                onClick={[Function]}
                                type="button"
                              >
                                Login with Facebook
                              </button>
                            </span>
                          </t>
                        </t>
                      </center>
                    </div>
                    <br />
                    <Divider
                      horizontal={true}
                    >
                      <div
                        className="ui horizontal divider"
                      >
                        Or
                      </div>
                    </Divider>
                    <div
                      className="spaceComponents"
                    >
                      <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/signup"
                      >
                        <Route
                          path="\\\\/signup"
                        >
                          <Link
                            aria-current={null}
                            replace={false}
                            to="/signup"
                          >
                            <a
                              aria-current={null}
                              href="/signup"
                              onClick={[Function]}
                            >
                              <Header
                                padded="very"
                                style={
                                  Object {
                                    "color": "#008080",
                                  }
                                }
                                textAlign="center"
                              >
                                <div
                                  className="ui center aligned header"
                                  padded="very"
                                  style={
                                    Object {
                                      "color": "#008080",
                                    }
                                  }
                                >
                                  Sign up
                                </div>
                              </Header>
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </div>
                  </div>
                </Segment>
              </div>
            </GridColumn>
          </div>
        </Grid>
      </div>
    </Login>
  </Router>
</BrowserRouter>
`;
